pilot/pkg/xds/authentication.go

```diff
@@ -31,7 +31,7 @@ import (
 // If no authenticators are configured, or if the request is on a non-secure
 // stream ( 15010 ) - returns an empty list of principals and no errors.
 func (s *DiscoveryServer) authenticate(ctx context.Context) ([]string, error) {
-	if !features.XDSAuth {
+	if !features.XDSAuth || len(s.Authenticators) == 0 {
```

Comment:
Why would we have `XDSAuth` enabled and not have any authenticators? That seems like a misconfiguration that shouldn't silently succeed

Reply:
Revised to return "XDSAuth is true but there is no authenticator" when XDSAuth is true but there is no authenticator.
With this code, when XDSAuth is true but there is no authenticator, the authenticate() will not have the following behavior:
	adsLog.Error("Failed to authenticate client from ", peerInfo.Addr.String(), "\n", strings.Join(authFailMsgs, "; "))
	return nil, errors.New("authentication failure")